src/networks/competetive_net.py

```diff
@@ -238,4 +262,217 @@ def compute_loss_against(self, opponent, input):
         outputs = sm(outputs_full[:, -1, :].contiguous().view(-1))
         d_loss_fake = self.loss_function(outputs, fake_labels)
 
-        return d_loss_real + d_loss_fake, None
+        return d_loss_real + d_loss_fake, None, None
+
+
+class SSDiscriminatorNet(DiscriminatorNet):
+
+    def __init__(self, label_pred_loss, num_classes, net, classification_layer, data_size,
+                 optimize_bias=True, mnist_28x28_conv=False):
+        DiscriminatorNet.__init__(self, label_pred_loss, net, data_size, optimize_bias=optimize_bias)
+        self.num_classes = num_classes
+        self.classification_layer = classification_layer.cuda() if is_cuda_enabled() else classification_layer
+        self.mnist_28x28_conv = mnist_28x28_conv
+
+        cc = ConfigurationContainer.instance()
+        self.in_mean = cc.settings['network'].get('in_mean', 0.0)
+        self.in_std = cc.settings['network'].get('in_std', 1e-10)
+        self.in_std_decay_rate = cc.settings['network'].get('in_std_decay_rate', 0.0)
+        self.in_std_min = cc.settings['network'].get('in_std_min', 1e-10)
+        self.in_fake_decay = cc.settings['network'].get('in_fake_decay', False)
+        self.label_rate = cc.settings['dataloader'].get('label_rate', 1)
```

Comment:
I think that the names of the variables should change since it is a little messy to understand the process.